File: server/router/miser/miser_transaction_item.go
package miser

import (
	"github.com/gin-gonic/gin"
	"github.com/springbear2020/self-hub/server/middleware"
)

type MiserTransactionItemRouter struct{}

func (s *MiserTransactionItemRouter) InitMiserTransactionItemRouter(Router *gin.RouterGroup, PublicRouter *gin.RouterGroup) {
	miserTransactionItemRouter := Router.Group("miserTransactionItem").Use(middleware.OperationRecord())
	{
		miserTransactionItemRouter.POST("createMiserTransactionItem", miserTransactionItemApi.CreateMiserTransactionItem)
		miserTransactionItemRouter.DELETE("deleteMiserTransactionItem", miserTransactionItemApi.DeleteMiserTransactionItem)
		miserTransactionItemRouter.DELETE("deleteMiserTransactionItemByIds", miserTransactionItemApi.DeleteMiserTransactionItemByIds)
		miserTransactionItemRouter.PUT("updateMiserTransactionItem", miserTransactionItemApi.UpdateMiserTransactionItem)
	}

	miserTransactionItemRouterWithoutRecord := Router.Group("miserTransactionItem")
	{
		miserTransactionItemRouterWithoutRecord.GET("findMiserTransactionItem", miserTransactionItemApi.FindMiserTransactionItem)
		miserTransactionItemRouterWithoutRecord.GET("getMiserTransactionItemList", miserTransactionItemApi.GetMiserTransactionItemList)
		miserTransactionItemRouterWithoutRecord.GET("listItemDistinctNames", miserTransactionItemApi.ListItemDistinctNames)
	}

	miserTransactionItemRouterWithoutAuth := PublicRouter.Group("miserTransactionItem")
	{
		_ = miserTransactionItemRouterWithoutAuth
	}
}

File: web/src/api/miser/miser_transaction_item.js
import service from '@/utils/request'

export const createMiserTransactionItem = (data) => {
  return service({
    url: '/miserTransactionItem/createMiserTransactionItem',
    method: 'post',
    data
  })
}

export const deleteMiserTransactionItem = (params) => {
  return service({
    url: '/miserTransactionItem/deleteMiserTransactionItem',
    method: 'delete',
    params
  })
}

export const deleteMiserTransactionItemByIds = (params) => {
  return service({
    url: '/miserTransactionItem/deleteMiserTransactionItemByIds',
    method: 'delete',
    params
  })
}

export const updateMiserTransactionItem = (data) => {
  return service({
    url: '/miserTransactionItem/updateMiserTransactionItem',
    method: 'put',
    data
  })
}

export const findMiserTransactionItem = (params) => {
  return service({
    url: '/miserTransactionItem/findMiserTransactionItem',
    method: 'get',
    params
  })
}

export const getMiserTransactionItemList = (params) => {
  return service({
    url: '/miserTransactionItem/getMiserTransactionItemList',
    method: 'get',
    params
  })
}

export const listItemDistinctNames = () => {
  return service({
    url: '/miserTransactionItem/listItemDistinctNames',
    method: 'get'
  })
}